src/app/components/abstract/abstract-sortable-table/abstract-sortable-table.component.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { AbstractSortableTableComponent } from './abstract-sortable-table.component';
+import { SortService } from '@services/sort-service/sort-service';
+import { IAbstractSortState } from './interfaces/abstract-sortable-table-interfaces';
+import { ABSTRACT_EXISTING_SORT_STATE_MOCK } from './mocks/abstract-sortable-table-existing-sort-state-mock';
+import { MOCK_ABSTRACT_TABLE_DATA } from './mocks/abstract-sortable-table-data-mock';
+
+class TestComponent extends AbstractSortableTableComponent {
+  constructor() {
+    super();
+    this.abstractTableData = MOCK_ABSTRACT_TABLE_DATA;
+    this.abstractExistingSortState = null;
+  }
+  public testOnSortChange(event: { key: string; sortType: 'ascending' | 'descending' }): void {
```

Comment:
What's the reason for this?

Reply:
Since i made the method protected i could not access it inside the testcomponent , so i decided to create a function to call it within the test file.